libcaf_net/caf/net/datagram_socket.hpp

```diff
@@ -0,0 +1,53 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2019 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#pragma once
+
+#include "caf/net/network_socket.hpp"
+#include "caf/variant.hpp"
+
+namespace caf {
+namespace net {
+
+/// A unconnected datagram-oriented network communication endpoint.
```

Comment:
I think `unconnected` is misleading here. If something has no notion of connections, than it also shouldn't be named unconnected IMO.

```suggestion
/// A datagram-oriented network communication endpoint.
```